### ISSUE TITLE
Two classes output only!

### ISSUE THREAD

**Issue by debayan** [BVLC/caffe#390]
I need a network with just two output classes A and B. I have tonnes of training data of images which either have A or B and are appropriately labelled. Once I have trained such a network, I wish to pass some images to test the network. Lets say I pass an image which contains neither A nor B. I understand that since the last layer is SOFTMAX it will still classify this image as either A or B with sum of probabilities of A and B equal to 1. Which means even if image does not have A it may still have a probability > 0.5 of carrying A according to the network.

Am I correct in my understanding? If yes, what kind of last layer should I use so that I can rely on a probability threshold to be able to reject such images which contain neither A not B.
